packages/core/src/types.ts

```diff
@@ -1854,21 +1856,36 @@ export type EventOfMachine<
 export interface ActorContext<TEvent extends EventObject, TSnapshot> {
   self: ActorRef<TEvent, TSnapshot>;
   name: string;
-  _event: SCXML.Event<TEvent> | LifecycleSignal;
+  sessionId: string;
+  logger: (...args: any[]) => void;
+  exec: ((fn: () => void) => void) | undefined;
```

Comment:
IMHO if the `ActorContext` is provided it already should have all the fields. In other words, `exec` should not be optional. If somebody doesn't want to actually execute things then can always provide a `noop`

Reply:
6691a6e8ee7444e25c095f2c946a88d213d0a4b9